app/controllers/withdraw_controller.rb

```diff
@@ -4,4 +4,11 @@ class WithdrawController < ApplicationController
   def create
     @document = Document.with_current_edition.find_by_param(params[:id])
   end
+
+  def new
+    document = Document.with_current_edition.find_by_param(params[:id])
+    public_explanation = params[:public_explanation]
+    UnpublishService.new.withdraw(document.current_edition, public_explanation)
+    redirect_to document
```

Comment:
The Unpublish Service checks that the edition is suitable for unpublishing so should be covered there https://github.com/alphagov/content-publisher/blob/bcdc97ebcb7aa6a6ce9d5c2467a9f71c6bad5430/app/services/unpublish_service.rb#L90

Reply:
I had a think about this - is this something we could add to the Withdrawal checker and display an error on the page if the edition is not suitable for withdrawing?

Reply:
@emmabeynon could you elaborate on this a bit? I'm wondering if you mean a catching an exception or not?

Reply:
I was thinking instead of catching an exception, displaying the error to the user like we do if there's a blank public explanation.  Raising an exception might be simpler though.